src/main/java/com/gl/donate_receive/service/converter/UserConverter.java

```diff
@@ -1,17 +1,30 @@
 package com.gl.donate_receive.service.converter;
 
-import com.gl.donate_receive.dto.UserCreateDto;
+
+import com.gl.donate_receive.dto.UserDto;
+import com.gl.donate_receive.model.Role;
 import com.gl.donate_receive.model.User;
+import lombok.AllArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Component;
 
 @Component
+@AllArgsConstructor
 public class UserConverter {
 
-	public User dtoToModel(UserCreateDto userCreateDto) {
+	private final PasswordEncoder passwordEncoder;
+
+	public User dtoToModel(UserDto userDto) {
 		return User.builder()
-			.login(userCreateDto.getLogin())
-			.password(userCreateDto.getPassword())
+			.login(userDto.getLogin())
+			.password(passwordEncoder.encode(userDto.getPassword()))
+			.role(getRole(userDto.getRole()))
 			.build();
 	}
+
+	private Role getRole(Role role) {
+		return role == null ? Role.USER : role;
```

Comment:
In general, I think there is no scenario where the user role should be null. We must ensure we don't store users without a role in DB. By doing so we might no longer need this method.

Reply:
Btw, since you've already started working on login we might need to implement a guest role as well. So that unregistered users will be able to see the catalog.

Reply:
> In general, I think there is no scenario where the user role should be null. We must ensure we don't store users without a role in DB. By doing so we might no longer need this method.

According to the existing user creation process, we create users without exact instructions about the user's Role while the signing process. And now UserDto we get from our frontend side without a specific user role. But in the existing case, we can set ROLE.USER in some different place for example...